fili-core/src/main/java/com/yahoo/bard/webservice/servlet/DelegatedPrincipal.java

```diff
@@ -0,0 +1,18 @@
+// Copyright 2022 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.servlet;
+
+import java.security.Principal;
+
+/**
+ * A subinterface to retrieve impersonator credentials from an impersonated principal.
+ */
+public interface DelegatedPrincipal extends Principal {
```

Comment:
Any particular reason we're making this an interface that we create an anonymous subclass for, instead of just making this a class that implements `Principal`? Making a class strikes me as being much clearer and just as effective for our purposes.